File: DataBinding/app/src/main/java/com/harshalbenake/databinding/MainActivity.java
package com.harshalbenake.databinding;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Toast;

import com.harshalbenake.databinding.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity implements Presenter{
    private int mCounter=0;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // Inflate the content view (replacing `setContentView`)
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        //adding data to pojoobject
        DataPojo dataPojo=new DataPojo("Harshal","harshalbenake@gmail.com");
        //setting datapojo to binding
        binding.setDataPojo(dataPojo);
        //set listner to9 binding
        binding.setPresenter(this);
        //setting textview appearance without creating textview object
        binding.tvName.setTextSize(25f);
    }

    @Override
    public void onSaveClick(View view, DataPojo dataPojo) {
        mCounter=mCounter+1;
        Toast.makeText(this,"Name: "+dataPojo.getName()+"\nEmail: "+dataPojo.getEmail()+"\n"+mCounter,Toast.LENGTH_SHORT).show();
    }


    /**
     * Inner static class Utils
     */
    public static class Utils {
        Context mContext;
        public Utils(Context context) {
            this.mContext=context;
        }

        /**
         * appends Surname
         * @param strText
         * @return
         */
        public static String appendSurname(String strText){
            return strText+" Benake";
        }
    }

}

File: KotlinDataBindingList/app/src/main/java/com/harshalbenake/kotlindatabindinglist/MainActivity.kt
package com.harshalbenake.kotlindatabindinglist

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import com.harshalbenake.kotlindatabindinglist.Adapters.PersonProfileAdapter
import com.harshalbenake.kotlindatabindinglist.Model.PersonProfile

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main)

        //array variable initialized
        var listItems = arrayListOf<PersonProfile>()
        //arraylist item added
        for (i in 0..10) {
            listItems.add(PersonProfile("Harshal Benake " + i, "harshalbenake" + i + "@gmail.com", 28 + i))
        }

        val recyclerView = findViewById<RecyclerView>(R.id.rv_personprofile)
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = PersonProfileAdapter(listItems)
    }
}
